crates/uv/src/commands/project/init.rs

```diff
@@ -173,33 +176,77 @@ async fn init_project(
         }
     };
 
-    // Add a `requires-python` field to the `pyproject.toml`.
-    let requires_python = if let Some(request) = python.as_deref() {
+    let reporter = PythonDownloadReporter::single(printer);
+    let client_builder = BaseClientBuilder::new()
+        .connectivity(connectivity)
+        .native_tls(native_tls);
+
+    // Add a `requires-python` field to the `pyproject.toml` and return the corresponding interpreter.
+    let (requires_python, python_request) = if let Some(request) = python.as_deref() {
         // (1) Explicit request from user
         match PythonRequest::parse(request) {
             PythonRequest::Version(VersionRequest::MajorMinor(major, minor)) => {
-                RequiresPython::greater_than_equal_version(&Version::new([
+                let requires_python = RequiresPython::greater_than_equal_version(&Version::new([
                     u64::from(major),
                     u64::from(minor),
-                ]))
+                ]));
+
+                let python_request = if no_pin_python {
+                    None
+                } else {
+                    Some(PythonRequest::Version(VersionRequest::MajorMinor(
+                        major, minor,
+                    )))
+                };
+
+                (requires_python, python_request)
             }
             PythonRequest::Version(VersionRequest::MajorMinorPatch(major, minor, patch)) => {
-                RequiresPython::greater_than_equal_version(&Version::new([
+                let requires_python = RequiresPython::greater_than_equal_version(&Version::new([
                     u64::from(major),
                     u64::from(minor),
                     u64::from(patch),
-                ]))
+                ]));
+
+                let python_request = if no_pin_python {
+                    None
+                } else {
+                    Some(PythonRequest::Version(VersionRequest::MajorMinorPatch(
+                        major, minor, patch,
+                    )))
+                };
+
+                (requires_python, python_request)
             }
-            PythonRequest::Version(VersionRequest::Range(specifiers)) => {
-                RequiresPython::from_specifiers(&specifiers)?
+            ref python_request @ PythonRequest::Version(VersionRequest::Range(ref specifiers)) => {
+                let requires_python = RequiresPython::from_specifiers(specifiers)?;
+
+                let python_request = if no_pin_python {
+                    None
+                } else {
+                    let interpreter = PythonInstallation::find_or_download(
```

Comment:
Not blocking, but do we do interpreter discovery twice in this case? Seems less than ideal if so.

Reply:
Nevermind doesn't look like it. Hard to see in the diff.